### PR TITLE
[Draft] Add integ test for pyxis + enroot

### PR DESCRIPTION
**Pull request by hehe7318**
### Description of changes
* Add integ test for pyxis + enroot

### Tests
* Ongoing, will update when finish.

### Checklist
- Make sure you are pointing to **the right branch**.
- If you're creating a patch for a branch other than `develop` add the branch name as prefix in the PR title (e\.g\. `[release-3.6]`).
- Check all commits' messages are clear, describing what and why vs how.
- Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integration-tests/tests/pyxis/test_pyxis.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+from assertpy import assert_that
+from remote_command_executor import RemoteCommandExecutor
+
+from tests.common.schedulers_common import SlurmCommands
+
+
+@pytest.mark.usefixtures("region", "os", "instance", "scheduler")
+def test_pyxis(pcluster_config_reader, clusters_factory, test_datadir):
+    """
+    Test Enroot and Pyxis failure due to concurrent sed operations on shared filesystem.
```
Comment (gmarciani) [inline tests/integration-tests/tests/pyxis/test_pyxis.py]: The goal of the test is to validate that Pyxis use case works, not to validate that it fail on a specific condition.
That said, we are interested in checking that it now fails on a specific race condition, but this is an internal consideration, not the goal of the test

File: tests/integration-tests/tests/pyxis/test_pyxis.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    job_id = slurm_commands.assert_job_submitted(result.stdout)
+    slurm_commands.wait_job_completed(job_id)
+
+    # Fetch the job output and check for the expected error messages
+    logging.info("Fetching job output")
+    slurm_out = remote_command_executor.run_remote_command("cat slurm.out").stdout
+
+    logging.info("Checking for expected error messages in job output")
+    assert_that(slurm_out).contains("Failed to open /opt/slurm/etc/plugstack.conf.d/")
```
Comment (gmarciani) [inline tests/integration-tests/tests/pyxis/test_pyxis.py:61]: We should do the other way around.
The test must verify that the job succeeds and in particular, that it does not contain this error in the output.

File: tests/integration-tests/tests/pyxis/test_pyxis/test_pyxis/pcluster.config.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        - Name: compute-resource-0
+          Instances:
+            - InstanceType: t3.small
+          MinCount: 0
+          MaxCount: 1000
+      Networking:
+        SubnetIds:
+          - {{ private_subnet_id }}
+SharedStorage:
+  - MountDir: {{ mount_dir }}
```
Comment (gmarciani) [inline tests/integration-tests/tests/pyxis/test_pyxis/test_pyxis/pcluster.config.yaml]: The EFs storage should be used as internal shared storage to replicate the race condition.
See https://docs.aws.amazon.com/parallelcluster/latest/ug/HeadNode-v3.html#yaml-HeadNode-SharedStorageType

**Review by gmarciani — CHANGES_REQUESTED** [review]

File: tests/integration-tests/tests/pyxis/test_pyxis.py (hunk 1 above)
Comment (hehe7318) [reply tests/integration-tests/tests/pyxis/test_pyxis.py]: I know, this test is only for current version, when Himani merge the fix PR, I will make changes to adapt it.

File: tests/integration-tests/tests/pyxis/test_pyxis.py (hunk 2 above)
Comment (hehe7318) [reply tests/integration-tests/tests/pyxis/test_pyxis.py:61]: I know, this test is only for current version, when Himani merge the fix PR, I will make changes to adapt it.

File: tests/integration-tests/tests/pyxis/test_pyxis.py (hunk 1 above)
Comment (hehe7318) [reply tests/integration-tests/tests/pyxis/test_pyxis.py]: For now it's only testing the pyxis+enroot will cause errors in current version.

Comment (hehe7318) [reply tests/integration-tests/tests/pyxis/test_pyxis.py]: I will not merge the test with current discription and implementation. I create this draft only for better troubleshooting.

File: tests/integration-tests/tests/pyxis/test_pyxis.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    assert_that(slurm_out).contains("Permission denied")
+    assert_that(slurm_out).contains("Plug-in initialization failed")
```
Comment (gmarciani) [inline tests/integration-tests/tests/pyxis/test_pyxis.py:63]: The first assertion is enough to prove the failure on the error message

File: tests/integration-tests/tests/pyxis/test_pyxis/test_pyxis/pcluster.config.yaml (hunk 3 above)
Comment (hehe7318) [reply tests/integration-tests/tests/pyxis/test_pyxis/test_pyxis/pcluster.config.yaml]: Fixed

**Pull request closed without merging**